src/libANGLE/renderer/d3d/DynamicHLSL.cpp

```diff
@@ -1400,6 +1445,21 @@ void BuiltinVaryingsD3D::updateBuiltins(gl::ShaderType shaderType,
         builtins->glPosition.enable(userSemantic, reservedSemanticIndex++);
     }
 
+    if (metadata.getRendererMajorShaderModel() >= 4 &&
+        (shaderType == gl::ShaderType::Vertex || shaderType == gl::ShaderType::Geometry))
```

Comment:
Shouldn't geometry shaders be just excluded?

Reply:
Well I was hoping to keep compatibility with geometry shaders but probably that would require more work to finish. Since we are not going to fix this extension for goemetry shaders (not it is required by the original spec.) it might cleaner to exclude it yes.